packages/server/src/core/procedure.ts

```diff
@@ -20,44 +20,24 @@ export interface ProcedureOptions {
 }
 
 /**
- * FIXME: this should only take 1 generic argument instead of a list
  * @internal
  */
-export interface ProcedureParams<
-  TConfig extends AnyRootConfig = AnyRootConfig,
-  TContextOut = unknown,
-  TInputIn = unknown,
-  TInputOut = unknown,
-  TOutputIn = unknown,
-  TOutputOut = unknown,
-  TMeta = unknown,
-> {
-  _config: TConfig;
-  /**
-   * @internal
-   */
-  _meta: TMeta;
-  /**
-   * @internal
-   */
-  _ctx_out: TContextOut;
-  /**
-   * @internal
-   */
-  _input_in: TInputIn;
-  /**
-   * @internal
-   */
-  _input_out: TInputOut;
-  /**
-   * @internal
-   */
-  _output_in: TOutputIn;
-  /**
-   * @internal
-   */
-  _output_out: TOutputOut;
-}
+type AnyProcedureParams = {
+  _config: AnyRootConfig;
+  _meta: unknown;
+  _ctx_out: unknown;
+  _input_in: unknown;
+  _input_out: unknown;
+  _output_in: unknown;
+  _output_out: unknown;
+};
+
+/**
+ * @internal
+ */
+export type ProcedureParams<
+  TParams extends AnyProcedureParams = AnyProcedureParams,
```

Comment:
Can we get rid of the default? Otherwise it's effectively the same as `AnyProcedureParams`

```suggestion
  TParams extends AnyProcedureParams,
```

Reply:
if I do that i get the following error, do you have an idea how I could possibly fix that?

<img width="1411" alt="CleanShot 2023-03-03 at 16 16 23@2x" src="https://user-images.githubusercontent.com/20271968/222771855-3da39904-2edf-4dad-8963-199ca13ce576.png">

Reply:
<img width="908" alt="CleanShot 2023-03-03 at 16 19 10@2x" src="https://user-images.githubusercontent.com/20271968/222772385-158113d1-4b56-4eed-bd54-c49c25073912.png">

adding the `AnyProcedureParams` as a generic does fix it but I'm not sure if that's what we are going for

Reply:
Yeah, we gotta be explicit with the generic, I think what you're doing here is fine!

Reply:
Okay I will update it to explicitly add the generic and will resubmit for review ❤️